client/src/services/roomService.ts

```diff
@@ -1,20 +1,38 @@
 import type { Room } from '../../types/roomType';
-import HttpHelper from '../utils/HttpHelper';
+import http from '../utils/HttpHelper';
+import { tr } from '../utils/i18nHelper';
+import toastService from '../utils/toast';
 
 export default class RoomService {
     static async getRooms(): Promise<Room[]> {
-        return HttpHelper.get<Room[]>('/room');
+        return http.get<Room[]>('/room')
+            .then((rooms) => rooms)
+            .catch(() => {
+                toastService.toast(tr('rooms.errorWhileLoading'), 'error');
+                return [];
+            });
     }
 
     static async createRoom(room: Room): Promise<Room> {
-        return HttpHelper.post('/room', room);
+        return http.post<Room, Room>('/room', room)
+            .then((r) => r)
+            .catch(() => {
+                toastService.toast(tr('rooms.errorWhileCreating'), 'error');
+                return room;
+            });
     }
 
     static async deleteRoom(room: Room): Promise<void> {
-        return HttpHelper.delete(`/room/${room.roomId}`);
+        return http.delete<void, Room>(`/room/${room.roomId}`)
+            .catch(() => {
+                toastService.toast(tr('rooms.errorWhileDeleting'), 'error');
+            });
     }
 
     static async updateRoom(room: Room): Promise<void> {
-        return HttpHelper.put(`/room/${room.roomId}`, room);
+        return http.put<void, Room>(`/room/${room.roomId}`, room)
+            .catch(() => {
+                toastService.toast(tr('rooms.errorWhileDeleting'), 'error');
```

Comment:
Is this update function equivalent to a delete?

Reply:
Nop, that's a typo